pkg/kubelet/images/image_gc_manager.go

```diff
@@ -223,12 +219,6 @@ func (im *realImageGCManager) GetImageList() ([]container.Image, error) {
 func (im *realImageGCManager) detectImages(ctx context.Context, detectTime time.Time) (sets.String, error) {
 	imagesInUse := sets.NewString()
 
-	// Always consider the container runtime pod sandbox image in use
-	imageRef, err := im.runtime.GetImageRef(ctx, container.ImageSpec{Image: im.sandboxImage})
```

Comment:
Here's where we need to interrogate the CRI impl for the sandbox image and insert that image into `imagesInUse`

If we don't do this, kubelet image GC discards pause image. We've seen this happen when the kubelet command line and containerd config.toml disagree on what the pause image is.

Reply:
kubelet currently respects the `Pinned` field in the ImageStatus response. If the CRI adds that field to the sandbox image, then it will be skipped as a candidate for GC

Reply:
Is this widely supported by all relevant versions of container runtimes?

Reply:
It's recently been added to [`CRI-O`](https://github.com/cri-o/cri-o/pull/6903), and both  [`Containerd`](https://github.com/containerd/containerd/pull/7944)  and [`cri-dockerd`](https://github.com/Mirantis/cri-dockerd/pull/194) are also planning to support it.

Reply:
@haircommander @sohankunkerkar until we have versions of all the runtimes support it, we can't land this.

Reply:
> @haircommander @sohankunkerkar until we have versions of all the runtimes support it, we can't land this.

:+1: I think that's reasonable.

Reply:
@dims : there are two different things, shown in two different places.

one is to show the pause image in the info, so that kubeadm can issue that warning (until it no longer cares)
`{{.config.sandboxImage}}`

the other is to make sure that the current pause image is reported as "pinned", in the image list / inspection
`{{.status.pinned}}`

both of which should be implemented for all runtimes, or at least "soon"

Reply:
Here was the CRI test that I was imagining:

```console
$ sudo crictl info --output go-template --template '{{.config.sandboxImage}}'
registry.k8s.io/pause:3.9
$ sudo crictl inspecti --output go-template --template '{{.status.pinned}}' registry.k8s.io/pause:3.9
true
```

But I am not sure where the real test lives?